### PR TITLE
An 4089/block rewards

### PR DESCRIPTION
**Pull request by tarikceric**
Build models for block rewards
- create 4 separate silver models for each reward type
- create `epoch` table using generator
- batch loading will be used to initially backfill the tables (via `dbt_run_batch_backfill` job). Once the backfill is caught up, that job will be turned off and models will be ran in regular incremental

### CODE REVIEW

File: models/silver/rewards/silver__fee_rewards.sql [hunk 1]
```diff
@@ -0,0 +1,150 @@
+{{ config(
+    materialized = 'incremental',
+    unique_key = ["vote_pubkey","epoch_earned","block_id"],
+    cluster_by = ['block_timestamp::DATE','floor(block_id,-6)','_inserted_timestamp::DATE'],
+    post_hook = "ALTER TABLE {{ this }} ADD SEARCH OPTIMIZATION",
```
Comment (eric-laurello) [inline models/silver/rewards/silver__fee_rewards.sql:5]: can you call out the columns that you want SO on? I don't think it needs to be on the amounts or timestamps
something like this:
`  post_hook = "ALTER TABLE {{ this }} ADD SEARCH OPTIMIZATION ON EQUALITY(msg_type, msg:attributes);",`

File: models/silver/rewards/silver__fee_rewards.sql [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        SELECT
+            MAX(_partition_id) -1
+        FROM
+            {{ this }}
+        )
+    AND _partition_id <= (
+        SELECT
+            MAX(_partition_id)
+        FROM
+        solana.streamline.complete_block_rewards
```
Comment (eric-laurello) [inline models/silver/rewards/silver__fee_rewards.sql:57]: replace the hard coded model

File: models/silver/rewards/silver__rent_rewards.sql [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        SELECT
+            MAX(_partition_id) -1
+        FROM
+            {{ this }}
+        )
+    AND _partition_id <= (
+        SELECT
+            MAX(_partition_id)
+        FROM
+        solana.streamline.complete_block_rewards
```
Comment (eric-laurello) [inline models/silver/rewards/silver__rent_rewards.sql:57]: replace the hard coded model

File: models/silver/rewards/silver__staking_rewards.sql [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        SELECT
+            MAX(_partition_id) -1
+        FROM
+            {{ this }}
+        )
+    AND _partition_id <= (
+        SELECT
+            MAX(_partition_id)
+        FROM
+        solana.streamline.complete_block_rewards
```
Comment (eric-laurello) [inline models/silver/rewards/silver__staking_rewards.sql:57]: replace the hard coded model

File: models/silver/rewards/silver__voting_rewards.sql [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+        SELECT
+            MAX(_partition_id) -1
+        FROM
+            {{ this }}
+        )
+    AND _partition_id <= (
+        SELECT
+            MAX(_partition_id)
+        FROM
+        solana.streamline.complete_block_rewards
```
Comment (eric-laurello) [inline models/silver/rewards/silver__voting_rewards.sql:57]: replace the hard coded model

File: models/silver/rewards/silver__fee_rewards.sql (hunk 1 above)
Comment (eric-laurello) [reply models/silver/rewards/silver__fee_rewards.sql:5]: since these are merges, we can add in:
` merge_exclude_columns = ["inserted_timestamp"],`

Comment (tarikceric) [reply models/silver/rewards/silver__fee_rewards.sql]: updated to include `merge_exclude_columns` and columns for SO

File: models/gold/gov/gov__dim_epoch.yml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+        tests:
+          - dbt_expectations.expect_column_to_exist
+      - name: START_BLOCK
+        description: "{{ doc('start_block') }}"
+        tests:
+          - dbt_expectations.expect_column_to_exist
+      - name: END_BLOCK
+        description: "{{ doc('end_block') }}"
+        tests:
+          - dbt_expectations.expect_column_to_exist
```
Comment (eric-laurello) [inline models/gold/gov/gov__dim_epoch.yml:17]: can you add the descriptions for the pk/ and timestamps to all the gold models?

**Review by eric-laurello — APPROVED** [review]

**Pull request merged**